test/asynchronous/test_client_context.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2018-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import os
+import sys
+
+sys.path[0:0] = [""]
+
+from test.asynchronous import AsyncUnitTest, SkipTest, async_client_context, unittest
+
+_IS_SYNC = False
+
+
+class TestAsyncClientContext(AsyncUnitTest):
+    @classmethod
+    async def _setup_class(cls):
+        pass
```

Comment:
Can we remove _setup_class since it's just a `pass`?

Reply:
`AsyncUnitTest` assumes that `_setup_class` would be implemented..without this the code would infinitely recurse and eventually error. 

To my knowledge this is the first time that a class inherits from `AsyncUnitTest` that doesn't "need" a `_setup_class`. I don't know if we want to change `AsyncUnitTest` to accommodate this class's use case or not...

Reply:
AsyncUnitTest should `pass` instead of trying to recursively call itself.

Reply:
That work's too, don't know why i didn't think of that earlier! Just changed :)